### DISCUSSION TITLE
Configure slot collection in the workflow

### DISCUSSION THREAD

**Discussion by YidaHu** [langgenius/dify#3350] [Suggestion]
### Self Checks

- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] Pleas do not modify this template :) and fill in all the required fields.

### 1. Is this request related to a challenge you're experiencing?

I can't configure slot collection. And I read the documentation. Nothing.

### 2. Describe the feature you'd like to see

I want to configure slot collection in the workflow, so that I can call a custom API based on the collected variables to handle the business. It is best to use an LLM to collect and generate reply.

thanks

### 3. How will this feature improve your workflow or experience?

For example: 
I want users to be able to provide a cell phone number in order to handle business.

### 5. Can you help us with this feature?

- [ ] I am interested in contributing to this feature.

**Comment by lileming** [comment]
I also have this confusion. Can the author add a node (or component) similar to a variable pool in the workflow, so that other nodes can read and write variable values from the variable pool at any time?

**Comment by patryk20120** [comment]
You can read variables at any time and you can create variables at any time? What is your use casa that you cannot figure out? Maybe I can help with an template workflow and export it for you?

**Reply by lileming** [reply]
I plan to use chat workflow to play the role of e-commerce customer service, which mainly involves collecting key messages through multiple rounds of dialogue with customers. After obtaining key information, recommend products to customers by calling APIs.
For example, when I work as a customer service representative for clothing, I can set variables such as height, weight, clothing color, and clothing style in the variable pool. At the beginning of the conversation, I will collect variable values from the variable pool through the conversation, and only after these variables are collected will I execute the subsequent nodes

**Reply by patryk20120** [reply]
"through the conversation" - you mean that the user has to type all the data?

**Reply by patryk20120** [reply]
Basically workflow is not meant to be interactive like chat. I think your solution is make a chatbot which will utilize workflow inside.

**Reply by lileming** [reply]
Yes, multiple rounds of conversation with the user to collect these variables
